File: Algorithms/01.Recursion/src/RecursiveFactorial.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class RecursiveFactorial {
    public static void main(String[] args) throws IOException {
        BufferedReader console = new BufferedReader(
                new InputStreamReader(System.in));

        int number = Integer.parseInt(console.readLine());

        int factorial = factorial(number);
        System.out.printf("Recursion solution of Factorial: %s",factorial);
    }
    private static int factorial(int number){
        if(number == 0){
            return  1;
        }

        return  number * factorial(number - 1);
    }
}

File: Algorithms/01.Recursion/Lab/02.RecursiveFactorial/RecursiveFactorial.cs
using System;

namespace Factorial
{
    public class RecursiveFactorial
    {
        public static void Main()
        {
            int n = int.Parse(Console.ReadLine());

            int resultFactorial = GetFactorial(n);
			Console.WriteLine($"{resultFactorial}");
        }

        public static int GetFactorial(int n)
        {
            if (n == 0)
            {
                return 1;
            }

            return n * GetFactorial(n - 1);
        }
    }
}
